left_sidebar: Make a collapsible "Views" section.

Ddharmani3 opened the pull request:
<!-- Describe your pull request here.-->
This is the first step toward the left sidebar redesign project.
This PR puts all the links in the top left area into a collapsible views section.

- [x]  Put all the links above "Direct messages" into a collapsible section labeled as "Views" (same font as "Direct messages" and "Streams"). The selected view should always be shown even when the section is collapsed (as we do for the selected DM conversation).

- [x] When the Views section is collapsed, show the links to views as icons after the "Views" label.

This PR does not redesign anything in the left sidebar all icons, colors etc. ar kept the same.


Fixes: #25902<!-- Issue link, or clear description.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
- Expanded views section: 
![image](https://github.com/zulip/zulip/assets/64723994/85ea588b-a5e5-492c-999e-011409809e81)

- Collapsed views section:
![image](https://github.com/zulip/zulip/assets/64723994/c65b15cd-122f-444c-8179-c8a62aee7e16)

- 3-dot menu when 3 options:
![image](https://github.com/zulip/zulip/assets/64723994/b2cd495d-6acf-40c3-8ce0-a3deb2e2338e)

- 3-dot menu when 4 options:
![image](https://github.com/zulip/zulip/assets/64723994/26131675-3956-4b23-92f2-e9dc365b3341)

- unread counter on hover:
![image](https://github.com/zulip/zulip/assets/64723994/e83fc24c-cec6-46f7-8579-67422413ac60)
- tooltips:
![image](https://github.com/zulip/zulip/assets/64723994/f735d275-aa6f-415d-a3c7-727dd00e8341)

- saved as drafts tooltip:
![image](https://github.com/zulip/zulip/assets/64723994/b09171cf-7850-4b03-9257-20bb50d9dbd6)

- Dark theme:
![image](https://github.com/zulip/zulip/assets/64723994/80efba53-ddee-41ad-b4e1-daca090a1c20)


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

Ddharmani3 commented:
@zulipbot add "buddy review"

Ddharmani3 commented:
@zulipbot add "QA needed"

terpimost commented:
In general I think functionally it does what was designed. If you guys want to see that on CZO to evaluate it, go ahead @timabbott 
I'm working on prettifying styles around such interactions. I don't think we should fit more than 5 items in collapsed version, right now I see 6, which makes the icon button too small to my taste.

Ddharmani3 commented:
> In general I think functionally it does what was designed. If you guys want to see that on CZO to evaluate it, go ahead @timabbott I'm working on prettifying styles around such interactions. I don't think we should fit more than 5 items in collapsed version, right now I see 6, which makes the icon button too small to my taste.

Issue #25902 mentions this: 
> To test and decide:
    - Does it look OK to have 6 icons shown (when the user has scheduled messages)? Or do we need to hide some buttons under a ... menu? (We definitely expect to need a ... menu once the Inbox view is added.)

Even I feel 6 Icons feel too cluttered. This is what it looks like 5 icons with the same font size (16px):

![image](https://github.com/zulip/zulip/assets/64723994/333b632f-18d4-49e1-9a62-e44bb7d25f4f)

And with a bigger font size(18px): 

![image](https://github.com/zulip/zulip/assets/64723994/9ce691ae-c006-45ab-99b9-1bd64afb86ad)

@terpimost should I add a 4px margin-bottom to collapsed views section to increase the gap between views and direct messages? also asked in details [here](https://github.com/zulip/zulip/pull/26035#issuecomment-1593545941) (point 3).

More feedback on 5 or 6 icons and font sizes can be taken from the community by test-deploying this PR.

terpimost commented:
> I removed the 8px margin-bottom in the collapsed state because there was excessive space between views and direct messages. However, they are now too close to each other. Should I add a 4px margin-bottom or another value?

Since this is not "designed" version. Make something that you feel is good and we will try it live
In this case I would add space between Views and DMs
<img width="417" alt="image" src="https://github.com/zulip/zulip/assets/1903309/759f9cf7-f265-42f1-9d11-fe6223c14500">

timabbott commented on web/src/ui_init.js:
```diff
@@ -104,7 +105,6 @@ import * as stream_settings_ui from "./stream_settings_ui";
 import * as sub_store from "./sub_store";
 import * as timerender from "./timerender";
 import * as tippyjs from "./tippyjs";
-import * as top_left_corner from "./top_left_corner";
```
This commit is almost certainly broken, since I don't see the new module being imported. Also looking at `git grep top_left_*`, I think we have a bunch of other things to rename in follow-up commits -- next time I'd do this sort of thing as a prep PR that just does the rename -- a lot easier to quickly review/merge that and it'll also catch mistakes like this for you.

In general these

timabbott replied:
Actually, it seems you did this correctly and I somehow misread; in any case, a good reason to do big renames in their own prep PRs is that it makes it easier to review/verify :).

timabbott commented on web/templates/left_sidebar.hbs:
```diff
@@ -1,6 +1,6 @@
-    <div class="narrows_panel">
-        <ul id="global_filters" class="filters">
+    <div id="left_sidebar_navigation_area">
+        <ul id="views" class="filters">
```
I don't think an ID as short as `views` is going to make us happy. How about `left_sidebar_navigation_list` for this element, since it's the list for that area.

timabbott commented on web/templates/left_sidebar.hbs:
```diff
@@ -64,10 +64,13 @@
                     </span>
                     {{~!-- squash whitespace --~}}
-                    <span>{{t 'Scheduled messages' }}</span>
+                    <span class="top_left_texts">{{t 'Scheduled messages' }}</span>
```
This class name seems poor too; we're trying to get rid of the `top_left` class of name. `left-sidebar-navigation-label` might be OK.

timabbott commented:
Thanks for your work on this @Ddharmani3!

I merged cd24620c4a5c09ee021715e91180c26d7b6e3aac for the first commit; the next few commits need changes to the names of the things you're refactoring in the theme of the ideas above.

timabbott commented:
Overall looks pretty close to the spec. A few UI issues I note. 
* One is that the transparency for the counts looks bad:
* ![image](https://github.com/zulip/zulip/assets/2746074/7ae6725a-23b4-49c2-92a1-3d1df4da7b40)
I think the color might be fine, but seeing the grey dot below it looks off.
* The `\vdots` menu doesn't have the width it needs for its click area; look at how we do that for other `\vdots` menus -- narrow icons without a square container have really small hover/click areas and that needs to be addressed.
* This may be a preexisting thing deserving a prep commit, but at least in the dark theme, the "BACK TO STREAMS" label doesn't seem to have the same color as `STREAMS`, and that's probably a bug?

karlstolley commented:
@Ddharmani3, let me know when you've addressed Tim's comments and resolved the new conflicts here so I can give this a full review.

Ddharmani3 commented:
Addressed feedback by @timabbott and updated the PR.

Also, rebased the PR to solve the conflicts in `left_sidebar.hbs` that arose after #26468 got merged, before updating this PR I also added a new class on span containing label names similar to #26468 which names them `global-filter-name` whereas I named them `left-sidebar-navigation-label` as suggested [here](https://github.com/zulip/zulip/pull/26035#discussion_r1272698335). The PR #26468 also adds a class `global-filter-container` to the `<a>` containing these `global-filter-name` spans.

So, In the updated PR, I renamed `global-filter-name` to `left-sidebar-navigation-label` and `global-filter-container` to `left-sidebar-navigation-label-container` in commit 07e8554aafe0878b8f3217c8f80da9366031b355.

@karlstolley This PR is ready for your review now.

karlstolley commented on web/e2e-tests/stars.test.ts:
```diff
@@ -28,7 +28,7 @@ async function toggle_test_star_message(page: Page): Promise<void> {
 
 async function test_narrow_to_starred_messages(page: Page): Promise<void> {
-    await page.click('#global_filters a[href^="#narrow/is/starred"]');
+    await page.click('#left_sidebar_navigation_list a[href^="#narrow/is/starred"]');
```
Let's rewrite `#left_sidebar_navigation_list` to [use hyphens](https://zulip.readthedocs.io/en/latest/subsystems/html-css.html#use-clear-unique-names-for-classes-and-object-ids). Same for any other classes or IDs created in this PR.

karlstolley commented on web/src/left_sidebar_navigation_area.js:
```diff
@@ -119,4 +146,15 @@ function do_new_messages_animation($li) {
 export function initialize() {
     update_scheduled_messages_row();
+    $("body").on("mouseenter touchstart", "#left_sidebar_navigation_list li", () => {
+        if ($("#left_sidebar_navigation_area").hasClass("collapsed")) {
```
Could this be solved in CSS? I'm not a fan of calling up `.css()` in general, and since all of this logic is about checking classes and selectors, it seems like a CSS-only solution should be possible. Failing that, I'd prefer toggling a class that takes these z-index values.

Ddharmani3 replied:
I tried a CSS-only solution by making the z-index of the left sidebar greater than the z-index of the top bar in commit c5ab8b7f0467f90fe3623f8eeeb9e87a27c69216. Can you please confirm if this is the right approach or not?

karlstolley commented on web/src/popover_menus.js:
```diff
@@ -921,28 +922,30 @@ export function initialize() {
-            $popper.one("click", "#unstar_all_messages", () => {
-                starred_messages_ui.confirm_unstar_all_messages();
-                instance.hide();
+    function starred_messages_onMount(instance) {
```
This should be probably kept to snake_case--`starred_messages_on_mount`, but there might even be a better name for this function than that.

Ddharmani3 replied:
Changed names to `snake_case` for now. Here are some options for a better name:

1) starred_messages_handlers
2) starred_messages_sidebar_menu_options
3) starred_messages_sidebar_menu_options_handler

karlstolley commented on web/src/popover_menus.js:
```diff
@@ -956,59 +959,96 @@ export function initialize() {
         },
     });
 
+    function drafts_sidebar_onMount(instance) {
```
Again, let's stick to snake_case.

karlstolley commented on web/src/popover_menus.js:
```diff
@@ -956,59 +959,96 @@ export function initialize() {
         },
     });
 
+    function all_messages_onMount(instance) {
```
This instance, too: `all_messages_on_mount`

karlstolley commented on web/styles/dark_theme.css:
```diff
@@ -510,6 +510,12 @@
     }
 
+    #left_sidebar_navigation_area.collapsed {
+        & li:not(:hover) .unread_count {
```
Can you explain why `:not(:hover)` is needed here?

karlstolley commented on web/styles/left_sidebar.css:
```diff
@@ -59,10 +59,36 @@ $before_unread_count_padding: 3px;
+
+#views_label_container {
+    cursor: pointer;
+    padding: 0 10px 0 6px;
```
It. would be good to document the origin of all of these different pixel values, here and throughout.

karlstolley commented on web/styles/left_sidebar.css:
```diff
@@ -171,14 +197,98 @@ li.show-more-topics {
+#left_sidebar_navigation_area {
+    display: flex;
+    width: 100%;
+    flex-direction: column;
```
Can you explain why this needs to be a column flex, and not just a regular content box (e.g., `display: block`)?

karlstolley commented on web/templates/left_sidebar.hbs:
```diff
@@ -1,69 +1,76 @@
-    <div class="narrows_panel">
-        <ul id="global_filters" class="filters">
+    <div id="left_sidebar_navigation_area">
+        <div id="views_label_container">
```
Just to reiterate: these new classes and IDs should be separated with hyphens, not underscores.

karlstolley commented:
Okay @Ddharmani3 -- I've pointed out some things with just a review of the code. I still need to pull this down and poke at it, however.

Ddharmani3 commented:
Updated the PR to address feedback and cleaned up CSS. Feedback is required on the following decisions:

1) I have limited the number of views in collapsed views to the first 5 only as 5+ views were feeling too cluttered.

2) Due to restricting the number of views to 5 and the recent addition of the Inbox view. In the collapsed state, we can't see the count of drafts or the count of scheduled messages(if any).

3) This is related to the 2nd Issue, when any message is `saved as draft` we don't get any indication as the drafts view is not present in collapsed state. Before the Inbox view this is how it was showing: 
![image](https://github.com/zulip/zulip/assets/64723994/628a3775-6313-4c28-94f3-9f6c661916dd)
In the mobile view, when the left sidebar is collapsed in the hamburger menu, we don't get any indication that time either, so not sure if this is a big Issue or not?

4) To prevent the unread counter from being cut by the top bar, I increased the z-index of the left sidebar to be higher than the top bar(103>102). Is it the right approach? It is not affecting the UI or layout of the app as far as I noticed. also asked this [here](https://github.com/zulip/zulip/pull/26035#discussion_r1331195428).
without Increasing: 
![image](https://github.com/zulip/zulip/assets/64723994/62a3949a-e738-4796-a1a2-335965a02cfd)

after Increasing:
![image](https://github.com/zulip/zulip/assets/64723994/11f94748-cebe-4731-a02f-b82cf8a81377)

5) In the portico/logged-out user screen, I have not implemented this as it only has 2 views.
![image](https://github.com/zulip/zulip/assets/64723994/95391a40-3fe3-41e6-bc8f-1897e7cf3a22)

Design changes will be in follow-up PRs. These are the final product screenshots after the PR update:
<details>
  <summary>Uncollapsed view</summary>
![image](https://github.com/zulip/zulip/assets/64723994/eb5a3eb2-7fdb-4cf1-8e5c-57acd19d6a5a)
</details>

<details>
  <summary>Collapsed view</summary>
![image](https://github.com/zulip/zulip/assets/64723994/94277901-c937-43cb-9832-efd8d137fb54)
![image](https://github.com/zulip/zulip/assets/64723994/e2c6b329-d89d-421d-b355-b62eb286188b)
![image](https://github.com/zulip/zulip/assets/64723994/7c9f459e-9b1d-4165-99cf-9620da6f26d0)

3-dot Menu with 3 options:
![image](https://github.com/zulip/zulip/assets/64723994/8143def7-3555-4443-b6b7-3b581944610b)

3-dot menu with 4 options:
![image](https://github.com/zulip/zulip/assets/64723994/42bd8081-d8d2-448a-b784-7a272df24111)

</details>

karlstolley commented:
Some other thoughts: 

* The vdots menu seems surprising and suboptimal to me. I get that it basically collects all of the different vdots options from the expanded Views area, but having all of those together in a basically decontextualized list is really kind of jarring. Like, it makes me question what it is that I'm looking at with those collapsed icons.
* The flexboxes as you've defined them need some work. There are bits of margin and padding that are blowing those open in unpredictable ways. I can work on that, if you'd like. I see this as leading the way of how other elements in the left sidebar will be designed as flexboxes, so it's important we do a really great, example-setting job here.
* We should get @terpimost (Vlad's) new icons placed for this, rather than the legacy Font Awesome ones. We should also find out if the new star icons in use on the hover controls can/should be used here.

Ddharmani3 commented:
Updated the PR to resolve conflicts and Improved some CSS. Addressed some feedback:

> We should get @terpimost (Vlad's) new icons placed for this, rather than the legacy Font Awesome ones. We should also find out if the new star icons in use on the hover controls can/should be used here.

Did this for the star icon in the last commit e5b6434f52ce4e3949639f031830905f44909a1e. There were no other views for which we currently use a new icon instead of the legacy Font Awesome ones.

>  But I would very much like to imagine the dot as the very center or at least near-center of the numbered version--almost as though on hover the numbers are popping out of that dot.

>That looks off to me, and crowded. Also the outline on starred messages shows the top navbar border shining through:

Updated the `.unread_counter` to feel like they are popping out of dots on hover. For smaller numbers, it feels as if the dot is on the very center, and for large numbers, they feel near-centered. Just as you mentioned :).
Also, after the updated CSS, the unread counters are not being cut by the header, so there is no need for `z-index: 103` for `.left-sidebar`.

<details>
  <summary>New unread counter behaviour and position</summary>

![image](https://github.com/zulip/zulip/assets/64723994/6a5a5057-8e39-45f7-a60a-2d65fb09d179)
![image](https://github.com/zulip/zulip/assets/64723994/8e215986-87b0-4995-ad2a-cb458a09136a)
![image](https://github.com/zulip/zulip/assets/64723994/07064ccf-7fec-40a5-afc4-50d7e76bbc1e)

https://github.com/zulip/zulip/assets/64723994/a46cd4dd-457a-4b06-8049-18317d929084

</details>

@terpimost Though this PR is only for functional change, and design changes will be another pass, but for now, do you have some simpler design solution for the transparent unread counters as they look very off to both Tim and Karl?
![image](https://github.com/zulip/zulip/assets/64723994/bf880282-88e5-46b5-ab25-67c0434b4c6d)


Awaiting responses on other feedback on CZO: https://chat.zulip.org/#narrow/stream/101-design/topic/Make.20views.20section.20collapsible.20feedback

Ddharmani3 commented:
@karlstolley, Updated the PR to address design feedback from [CZO](https://chat.zulip.org/#narrow/stream/101-design/topic/Make.20views.20section.20collapsible.20feedback/near/1646524) by @alya.

Here's the updated design screenshot: 
![image](https://github.com/zulip/zulip/assets/64723994/334ec2c8-6828-4d9b-8ab9-fa2e0d08cec0)
![image](https://github.com/zulip/zulip/assets/64723994/3cfd4751-2360-41c3-90be-d56e34006435)

![image](https://github.com/zulip/zulip/assets/64723994/b5e21b81-5258-4448-9b13-3bcebe9cf5c6)

@alya can you please do a manual testing if the new design matches your feedback?

Also, One more small piece of feedback is needed, as the combined 3-dot menu is a left sidebar popover menu, I have given `left_sidebar_tippy_options` to the 3-dot popover menu which has placement `right` by default with `bottom` fallback placement.

Currently, the design gets the `bottom` fallback placement and looks like this:
![image](https://github.com/zulip/zulip/assets/64723994/2ad8e929-4e53-466c-a8a2-37e2aa59664a)

should I change it to `right` always to make it look like this:
![image](https://github.com/zulip/zulip/assets/64723994/7f501444-4c66-4ab2-b61d-19221a637af0)

which orientation of the combined 3-dot menu looks better?

karlstolley commented on web/src/left_sidebar_navigation_area.js:
```diff
@@ -52,6 +54,31 @@ export function deselect_top_left_corner_items() {
     remove($(".top_left_inbox"));
 }
 
+function expand() {
```
A future improvement here would be to just add/remove the `collapsed` class and better handle the icon logic from CSS. But I don't see any benefit to fixing that now. Just something to note.

Ddharmani3 replied:
Sure once this PR gets merged, will create a follow-up PR that will handle the logic using CSS for this and also for the icon for direct messages as currently both same share logic.

karlstolley commented on web/src/popover_menus.js:
```diff
@@ -983,59 +987,105 @@ export function initialize() {
         },
     });
 
+    function all_messages_on_mount(instance) {
```
I'd kind of like to see all of the `_on_mount` functions defined together, and then keep the various `register_` functions grouped, too. I think that'll be easier for future contributors to scan.

karlstolley commented on web/src/popover_menus.js:
```diff
@@ -983,59 +987,105 @@ export function initialize() {
+            instance.setContent(parse_html(content));
+        },
+        onMount(instance) {
+            starred_messages_on_mount(instance);
```
Is there a reason not to have a named function for this onMount instance, too?

karlstolley commented on web/styles/compose.css:
```diff
@@ -802,6 +802,15 @@ input.recipient_box {
+            margin-left: 3px;
+
+            /* Dont show count on smaller widths due to space restrictions. */
+            @media (width < calc( $md_min + 32px)) {
```
I'm not a fan of this `calc()` bit, because then the list of breakpoints no longer really describes where they layout might shift. If there's not enough space at `md_min`, let's either include another breakpoint or just accept things as they are for now and shift to a larger breakpoint here.

karlstolley commented on web/styles/dark_theme.css:
```diff
@@ -779,7 +779,7 @@
         .show-all-streams {
-            color: hsl(236deg 33% 90%);
+            color: hsl(0deg 0% 100% / 75%);
             opacity: 0.75;
```
Can we strike this separate `opacity` property, since presumably that's now being expressed with the `hsl()` function?

karlstolley commented on web/styles/left_sidebar.css:
```diff
@@ -224,14 +219,115 @@ li.show-more-topics {
 
-        &:hover {
+#left-sidebar-navigation-area {
+    & li.top_left_row {
```
Once the rows are all gridded, which I'm working on, this CSS will need to be revisited. Just another thing to note for future improvements.

karlstolley commented on web/styles/left_sidebar.css:
```diff
@@ -224,14 +219,115 @@ li.show-more-topics {
+        margin-bottom: 2px;
+
+        /* Only show first 5 views in collapsed state. */
+        .top_left_row:nth-child(n + 6) {
```
I'm not a fan of this approach, because it leaves the visible views up to source order in the HTML. I think it would be better to just create a new class for the views that aren't shown in the collapsed state (`hide-in-collapsed-state`), and then with any lucky, that additional class will mean you don't need to resort to using `!important` here.

karlstolley commented on web/styles/left_sidebar.css:
```diff
@@ -224,14 +219,115 @@ li.show-more-topics {
+            margin-bottom: 0;
+            padding: 0;
+
+            & li {
```
Let's use the `top_left_row` class for now (I think it and `bottom_left_row` should be rewritten during the redesign, but let's worry about that then).

karlstolley commented on web/styles/left_sidebar.css:
```diff
@@ -224,14 +219,115 @@ li.show-more-topics {
+                            }
+
+                            &.zulip-icon-star-filled {
+                                top: 1px;
```
Overall you've done a tremendous job using flexbox in this pass (as of my review on 2023-09-25). In a follow-up PR, probably, I'd like to get rid of the fiddly little bits of positioning on icons like this. If we get the boxes right around the icons, then the flexbox's `align-content: center` should handle this.

karlstolley commented on web/styles/left_sidebar.css:
```diff
@@ -224,14 +219,115 @@ li.show-more-topics {
+            }
+        }
+
+        /* CSS to align unread counters in top right corner. */
```
Great job with these.

karlstolley commented on web/styles/left_sidebar.css:
```diff
@@ -224,14 +219,115 @@ li.show-more-topics {
+        }
+
+        /* Until hovered on li, show unread counters as dots. */
+        & li:not(:hover) .unread_count {
```
Can we go the other direction and use the `:hover` element on the `.unread_count` above? That will better reflect the state of things in the stylesheet, and not make people's heads ache trying to parse `:not(:hover)`

karlstolley commented on web/styles/left_sidebar.css:
```diff
@@ -224,14 +219,115 @@ li.show-more-topics {
+            top: auto;
+            /* 12px margin-right of #stream_filters to align with other vertical 3-dots in left sidebar. */
+            margin-right: 12px;
+            color: var(--color-global-filter-icon);
```
Since you've (thankfully) gotten rid of the `global_filters` references in this PR, we should rename it (I think this is also referenced around line 62 above). 

For the vdots here, I'm thinking you ought to specify something like `--color-vdots-navigation-menu`, and just assign it to the renamed global-filter icon style to keep those in sync.

Hopefully in the redesign, that additional vdots variable becomes unnecessary and we can use one of the existing vdots colors so we don't go back to having so many different colors for them.

karlstolley commented on web/styles/zulip.css:
```diff
@@ -3316,3 +3319,15 @@ select.invite-as {
+}
+
+.extra-views-menu .unread_count {
+    background-color: hsl(200deg 100% 40%);
```
So I was just about to ask about the new `--color-background-unread-counter` and whether it applies to all of the unread counts app-wide. This shows here that it doesn't. Two points there:

1. Rename the variable for its more specific context, I'm thinking.
2. Set up another variable here for this color, so we can track and apply it elsewhere as needed.

karlstolley commented on web/templates/compose_banner/success_message_scheduled_banner.hbs:
```diff
@@ -3,7 +3,7 @@
     <div class="main-view-banner-elements-wrapper {{#if button_text}}banner-contains-button{{/if}}">
         <p class="banner_content">{{t 'Your message has been scheduled for {deliver_at}.' }}
-            <a href="#scheduled">{{t "View scheduled messages" }}</a>
+            <a href="#scheduled">{{t "View {scheduled_messages_count} scheduled messages" }}</a>
```
I wonder if we have any kind of pluralize function or filter on the frontend?

karlstolley commented:
Whew. Closer than ever, @Ddharmani3. See my comments here, most of which I hope should be pretty easy to clean up. Once those are done, let's get Tim's review and see what he thinks about some burn-in time on CZO. I'm hopeful we can get to that point this week. Let me know if there's anything I can clarify here or any way that I can further help you out.

karlstolley commented on web/src/left_sidebar_navigation_area.js:
```diff
@@ -1,12 +1,53 @@
 
 let last_mention_count = 0;
 let left_sidebar_navigation_area_expanded = true;
+const views_data = {
```
Yeah, I've been playing with and thinking about this, but it's not the way to go. This introduces a whole lot of tricky code duplication with the ordinary rows in `left_sidebar.hbs`.

I think a better way to do this would be to create partials for each of these rows. The could then be loaded up at compile time within `left_sidebar.hbs`, but also then accessed independently for the active row.

That said, there are some other structural things I'm pondering here, so I'll follow back up if my thinking on this changes.

timabbott commented:
Closing as https://github.com/zulip/zulip/pull/27190 was just merged, completing this work.
